src/dsync/dsync.c

```diff
@@ -3188,7 +3190,7 @@ int main(int argc, char **argv)
     }
 
     /* Generate default output */
-    if (list_empty(&options.outputs)) {
+    if (list_empty(&options.outputs) && ret == 0) {
```

Comment:
Can we just clean up and `exit()` up above, at
```
3147         case 'o':
3148             ret = dsync_option_output_parse(optarg, 0);
3149             if (ret) {
3150                 usage = 1;
3151             }
3152             break;
```

If we're going to show the usage, then apparently the user provided invalid options, and we shouldn't waste time doing anything else, IMHO.

Also, right now nothing else above uses the `ret` variable, but in the future someone might decide it's useful for processing another option and not realize it affects behavior down below.

Reply:
I think that as soon as a usage is set to 1 we should just break, cleanup and show usage. Are we on the same page?

Reply:
Yes, I agree with you